src/main/java/org/springframework/samples/petclinic/rest/VisitRestController.java

```diff
@@ -78,12 +78,13 @@
 	public ResponseEntity<Visit> addVisit(@RequestBody @Valid Visit visit, BindingResult bindingResult, UriComponentsBuilder ucBuilder){
 		BindingErrorsResponse errors = new BindingErrorsResponse();
 		HttpHeaders headers = new HttpHeaders();
-		if(bindingResult.hasErrors() || (visit == null) || (visit.getPet() == null)){
+		if(bindingResult.hasErrors() || (visit == null) || (visit.getPet() == null) || visit.getVet() == null){
 			errors.addAllErrors(bindingResult);
 			headers.add("errors", errors.toJSON());
 			return new ResponseEntity<Visit>(headers, HttpStatus.BAD_REQUEST);
 		}
-		this.clinicService.saveVisit(visit);
+		visit.setPaid(false);
+        this.clinicService.saveVisit(visit);
```

Comment:
Consider moving it to service, as need additional checks on constraints like if that time avaliable. May be in future we would need to sending payment bill to client and other things not related to controller.